src/main/kotlin/build/buf/gradle/FormatConfiguration.kt

```diff
@@ -27,18 +27,41 @@ internal fun Project.configureFormat() {
 }
 
 private fun Project.configureBufFormatCheck() {
-    registerBufTask<FormatCheckTask>(BUF_FORMAT_CHECK_TASK_NAME) {
+    registerBufExecTask<FormatCheckTask>(BUF_FORMAT_CHECK_TASK_NAME) {
         group = VERIFICATION_GROUP
         description = "Checks that a Protobuf schema is formatted according to Buf's formatting rules."
         enabled = getExtension().enforceFormat
+
+        inputFiles.setFrom(
+            fileTree(workingDir.get()) {
+                include("**/*.proto")
+                // not to interfere with random plugins producing output to build dir
+                exclude("build")
+            },
+        )
     }
 
     tasks.named(CHECK_TASK_NAME).dependsOn(BUF_FORMAT_CHECK_TASK_NAME)
 }
 
 private fun Project.configureBufFormatApply() {
-    registerBufTask<FormatApplyTask>(BUF_FORMAT_APPLY_TASK_NAME) {
+    registerBufExecTask<FormatApplyTask>(BUF_FORMAT_APPLY_TASK_NAME) {
         group = VERIFICATION_GROUP
         description = "Formats a Protobuf schema according to Buf's formatting rules."
+
+        inputFiles.setFrom(
+            fileTree(workingDir.get()) {
+                include("**/*.proto")
+                // not to interfere with random plugins producing output to build dir
+                exclude("build")
+            },
```

Comment:
This seems to be a pretty common code block. Do you anticipate it changing across invocations, or can it be extracted to a helper function somewhere?

Reply:
Extracted to a function in the last commit.